src/main/java/ch/njol/skript/conditions/CondCanSee.java

```diff
@@ -1,85 +1,70 @@
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
 
 @Name("Can See")
-@Description("Checks whether the given players can see another players.")
-@Examples({"if the player can't see the player-argument:",
-		"\tmessage \"&lt;light red&gt;The player %player-argument% is not online!\""})
-@Since("2.3")
+@Description("Checks whether the given players can see the provided entities.")
+@Examples({
+	"if sender can't see the player-argument:",
+		"\tmessage \"who dat?\"",
+	"",
+	"if the player can see the last spawned entity:",
+		"\tmessage \"hello there!\""
+})
+@Since("2.3, INSERT VERSION (entities)")
+@RequiredPlugins("Minecraft 1.19+ (entities)")
 public class CondCanSee extends Condition {
 
 	static {
 		Skript.registerCondition(CondCanSee.class,
-				"%players% (is|are) [(1¦in)]visible for %players%",
-				"%players% can see %players%",
-				"%players% (is|are)(n't| not) [(1¦in)]visible for %players%",
-				"%players% can('t| not) see %players%");
+				"%entities% (is|are) [:in]visible for %players%",
+				"%players% can see %entities%",
+				"%entities% (is|are)(n't| not) [:in]visible for %players%",
+				"%players% can('t| not) see %entities%");
 	}
-	
+
 	@SuppressWarnings("null")
-	private Expression<Player> players;
+	private Expression<Player> viewers;
 	@SuppressWarnings("null")
-	private Expression<Player> targetPlayers;
+	private Expression<Entity> entities;
 
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult result) {
 		if (matchedPattern == 1 || matchedPattern == 3) {
-			players = (Expression<Player>) exprs[0];
-			targetPlayers = (Expression<Player>) exprs[1];
+			viewers = (Expression<Player>) exprs[0];
+			entities = (Expression<Entity>) exprs[1];
 		} else {
-			players = (Expression<Player>) exprs[1];
-			targetPlayers = (Expression<Player>) exprs[0];
+			entities = (Expression<Entity>) exprs[0];
+			viewers = (Expression<Player>) exprs[1];
 		}
-		setNegated(matchedPattern > 1 ^ parseResult.mark == 1);
+		setNegated(matchedPattern > 1 ^ result.hasTag("in"));
 		return true;
 	}
 
 	@Override
-	public boolean check(Event e) {
-		return players.check(e,
-				player -> targetPlayers.check(e,
+	public boolean check(Event event) {
+		return viewers.check(event,
+				player -> entities.check(event,
 						player::canSee
 				), isNegated());
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return PropertyCondition.toString(this, PropertyType.CAN, e, debug, players,
-				"see" + targetPlayers.toString(e, debug));
+	public String toString(@Nullable Event event, boolean debug) {
+		return PropertyCondition.toString(this, PropertyType.CAN, event, debug, viewers,
+				"see" + entities.toString(event, debug));
```

Comment:
I think this will need a space because otherwise it'll be `see...` instead of `see ...`. I guess that was just lacking in the original, I'm not sure.
```suggestion
				"see " + entities.toString(event, debug));
```